project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java

```diff
@@ -40,35 +42,66 @@
+    + " PRIMARY KEY (id) );";
+
+  private static String CREATE_INVESTMENTS_TABLE = 
+    "CREATE TABLE IF NOT EXISTS investments ( "
+    + "id SERIAL NOT NULL, user INT NOT NULL, competition INT NOT NULL, google_search VARCHAR(255) NOT NULL, "
+    + "invest_date DATE NOT NULL, sell_date DATE, amt_invested INT NOT NULL,"
+    + " PRIMARY KEY (id) );";
+
   private static final Logger log = Logger.getLogger(ConnectionPoolContextListener.class.getName());
   @SuppressFBWarnings(
       value = "USBR_UNNECESSARY_STORE_BEFORE_RETURN",
       justification = "Necessary for sample region tag.")
-  private DataSource createConnectionPool() {
+  public DataSource createConnectionPool() {
     // [START cloud_sql_mysql_servlet_create]
     // The configuration object specifies behaviors for the connection pool.
     HikariConfig config = new HikariConfig();
 
     log.info("I'm alive");
 
+    Config mySecrets = new Config();
+
     config.setJdbcUrl(String.format("jdbc:mysql://localhost:3306/%s", "test"));
     config.setUsername("root"); // e.g. "root", "mysql"
-    config.setPassword(REDACTED); // e.g. "my-password"
+    config.setPassword(mySecrets.databasePassword); // e.g. "my-password"
+
+    config.addDataSourceProperty("socketFactory", "com.google.cloud.sql.mysql.SocketFactory");
+    config.addDataSourceProperty("cloudSqlInstance",  "google.com:sgonks-step272:australia-southeast1:my-instance"); //"127.0.0.1");
 
+    log.info("I'm still alive");
     // Initialize the connection pool using the configuration object.
     return new HikariDataSource(config);
   }
 
-  private void createTable(DataSource pool) throws SQLException {
-    // Safely attempt to create the table schema.
+  private void createTables(DataSource pool) throws SQLException {
+    // Safely attempt to create users table
     try (Connection conn = pool.getConnection()) {
-      String stmt =
-          "CREATE TABLE IF NOT EXISTS users ( "
-              + "id SERIAL NOT NULL, name VARCHAR(255) NOT NULL, email VARCHAR(255) NOT NULL,"
-              + " PRIMARY KEY (id) );";
-      try (PreparedStatement createTableStatement = conn.prepareStatement(stmt);) {
-        createTableStatement.execute();
+      try (PreparedStatement createUsersStatement = conn.prepareStatement(CREATE_USERS_TABLE);) {
+        createUsersStatement.execute();
+      } try (PreparedStatement createCompetitionsStatement = conn.prepareStatement(CREATE_COMPETITIONS_TABLE);) {
+        createCompetitionsStatement.execute();
+      } try (PreparedStatement createParticipantsStatement = conn.prepareStatement(CREATE_PARTICIPANTS_TABLE);) {
+        createParticipantsStatement.execute();
+      } try (PreparedStatement createInvestmentsStatement = conn.prepareStatement(CREATE_INVESTMENTS_TABLE);) {
+        createInvestmentsStatement.execute();
       }
+      //get rid of this when we no longer need hard coded data
+      insertTestData(pool);
```

Comment:
Won't this create duplicate data on each startup? I would rather test data be added/removed manually.

Reply:
I've just been leaving it commented out, except when I change something in the database and need to drop all the tables and reinsert data.